src/Simulation/Common/IQubitManager.cs

```diff
@@ -3,41 +3,37 @@
 
 namespace Microsoft.Quantum.Simulation.Common
 {
-    using Microsoft.Quantum.Simulation.Core;
     using System.Collections.Generic;
+    using Microsoft.Quantum.Simulation.Core;
 
     public interface IQubitManager
     {
+        bool DisableBorrowing { get; }
+
+        bool IsValid(Qubit qubit);
+        bool IsFree(Qubit qubit);
+        bool IsDisabled(Qubit qubit);
+
+        long FreeQubitsCount { get; }
+        long AllocatedQubitsCount { get; }
+        IEnumerable<long> AllocatedIds();
+        long QubitsAvailableToBorrowCount(int stackFrame = 0);
```

Comment:
Having this as a method really breaks the symmetry. Is it really necessary?
Do we really need a method to pass the stackFrame? doesn't seem to be something that should live in the interface, maybe it can be a public method in the implementation?

Reply:
What symmetry? 
Before there were two methods, one for stackFrame = 0 and one for stackFrame = 1 that were needed as part of the interface. Now there is one method with an argument that makes it more general (I suspect at some point we may want higher up stack frames, so making it more general rather than  having it a separate method seems like a good choice to me.